### PR TITLE
Multi-dimensional indexes

### PR DESCRIPTION
**Pull request by maierlars**
Documentation PR for https://github.com/arangodb/arangodb/pull/13650

TODO:

- [x] Make the intended purpose clearer (i.e. few dimensions, an example use-case maybe)
- [x] A sentence or paragraph about what to expect from performance, i.e. YMMV.

### CODE REVIEW

File: 3.8/indexing-multi-dim.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        FILTER z0 <= p.z && p.z <= z1
+        RETURN p
+
+### Possible range queries
+
+Having an index on a set of fields does not require you to specify a full range
+for every dimension. For each dimension you can decide if you want to only bound
+it from one side (i.e. only an upper or lower bound) or not bound it at all.
+
+Futhermore you can use any comparsion operator. The index supports `<=` and `>=`
```
Comment (goedderz) [inline 3.8/indexing-multi-dim.md:56]:
```suggestion
Furthermore you can use any comparison operator. The index supports `<=` and `>=`
```

File: 3.8/indexing-multi-dim.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+naturally, `==` will be translated to the bound `[c, c]`. Strict comparsion
+is translated to they non-strict counterparts and a post-filter is inserted.
```
Comment (goedderz) [inline 3.8/indexing-multi-dim.md:58]:
```suggestion
is translated to their non-strict counterparts and a post-filter is inserted.
```

File: 3.9/indexing-multi-dim.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+and only if
+
+    a_2 <= b_1 and a_1 <= b_2
+
+Thus our query would be:
+
+    FOR app IN appointments
+        FILTER f <= app.to
+        FILTER app.from <= t
+        RETURN app
```
Comment (Simran-B) [inline 3.9/indexing-multi-dim.md:112]: Example incomplete?

Comment (maierlars) [reply 3.9/indexing-multi-dim.md:112]: No, I think its complete. What is missing? 🙂

File: 3.9/indexing-multi-dim.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        FILTER app.to <= t
+        RETURN app
+
+
+#### Finding all appointments that intersect a time range
+Given a time range `[f, t]` we want to find all appointments `[from, to]` that
+intersect `[f, t]`. Two intervals `[a_1, b_1]` and `[a_2, b_2]` intersect if
+and only if
+
+    a_2 <= b_1 and a_1 <= b_2
```
Comment (Simran-B) [inline 3.9/indexing-multi-dim.md:105]:
```suggestion
intersect `[f, t]`. Two intervals `[f, t]` and `[from, to]` intersect if
and only if

    f <= to and from <= t
```

File: 3.9/indexing-multi-dim.md [hunk 5]
```diff
@@ -64,13 +64,63 @@ is translated to their non-strict counterparts and a post-filter is inserted.
         FILTER p.z == 4
         RETURN p
 
+### Example Use Case
+
+If you build a calendar using ArangoDB you could create a collection for each user
+that contains her appointsments. The documents would roughly look as follows:
```
Comment (goedderz) [inline 3.9/indexing-multi-dim.md:70]:
```suggestion
that contains her appointments. The documents would roughly look as follows:
```

File: 3.9/indexing-multi-dim.md [hunk 6]
```diff
@@ -64,13 +64,63 @@ is translated to their non-strict counterparts and a post-filter is inserted.
...
+        "to": 678934,
+        "what": "Dentist",
+    }
+
+Where `from`/`to` are the timestamps when an appointment starts/ends. Having an
+multi-dimensional index on the fields
+
+    ["from", "to"]
+
+allows you to query effeciently for all appointsments within a given time range.
```
Comment (goedderz) [inline 3.9/indexing-multi-dim.md:83]:
```suggestion
allows you to query effeciently for all appointments within a given time range.
```

**Pull request merged**